fhir-server/src/test/java/com/ibm/fhir/server/test/FHIRRestHelperTest.java

```diff
@@ -0,0 +1,987 @@
+import com.ibm.fhir.model.resource.Practitioner;
+import com.ibm.fhir.model.resource.Practitioner.Qualification;
+import com.ibm.fhir.model.resource.Procedure;
+import com.ibm.fhir.model.type.Code;
+import com.ibm.fhir.model.type.CodeableConcept;
+import com.ibm.fhir.model.type.Coding;
+import com.ibm.fhir.model.type.Reference;
+import com.ibm.fhir.model.type.Uri;
+import com.ibm.fhir.model.type.code.BundleType;
+import com.ibm.fhir.model.type.code.EncounterStatus;
+import com.ibm.fhir.model.type.code.HTTPVerb;
+import com.ibm.fhir.model.type.code.ProcedureStatus;
+import com.ibm.fhir.persistence.FHIRPersistence;
+import com.ibm.fhir.server.util.FHIRRestHelper;
+
+public class FHIRRestHelperTest {
+
+    /**
+     * Test transaction bundle post single.
+     */
+    @Test
+    public void testTransactionBundlePostSingle() throws Exception {
+        FHIRPersistence persistence = new MockPersistenceImpl();
+        FHIRRestHelper helper = new FHIRRestHelper(persistence);
+        
+        Patient patient = Patient.builder().id("1").build();
+        Bundle.Entry.Request bundleEntryRequest = Bundle.Entry.Request.builder().method(HTTPVerb.POST).url(Uri.of("Patient")).build();
+        Bundle.Entry bundleEntry = Bundle.Entry.builder().resource(patient).request(bundleEntryRequest).build();
+        
+        Bundle requestBundle = Bundle.builder().id("bundle1").type(BundleType.TRANSACTION).entry(bundleEntry).build();
+
+        // Process bundle
+        FHIRRequestContext.get().setOriginalRequestUri("test");
+        Bundle responseBundle = helper.doBundle(requestBundle, null);
+        
+        // Validate results
+        assertNotNull(responseBundle);
+        assertEquals(1, responseBundle.getEntry().size());
+        Bundle.Entry entry = responseBundle.getEntry().get(0);
+        Bundle.Entry.Response response = entry.getResponse();
+        assertEquals("Patient/generated-0/_history/1", response.getLocation().getValue());
+        assertEquals(Integer.toString(Response.Status.CREATED.getStatusCode()), response.getStatus().getValue());
+    }
+
+    /**
+     * Test transaction bundle post with local reference dependency.
+     * Procedure has local reference to Patient.
+     */
+    @Test
+    public void testTransactionBundlePostWithDependency() throws Exception {
+        FHIRPersistence persistence = new MockPersistenceImpl();
+        FHIRRestHelper helper = new FHIRRestHelper(persistence);
+        
+        Patient patient = Patient.builder().id("1").build();
+        Bundle.Entry.Request bundleEntryRequest = Bundle.Entry.Request.builder().method(HTTPVerb.POST).url(Uri.of("Patient")).build();
+        Bundle.Entry bundleEntry = Bundle.Entry.builder().fullUrl(Uri.of("urn:1")).resource(patient).request(bundleEntryRequest).build();
+        
+        Procedure procedure = Procedure.builder().status(ProcedureStatus.COMPLETED).subject(Reference.builder().reference(com.ibm.fhir.model.type.String.of("urn:1")).build()).build();
```

Comment:
Building these test resources programmatically is totally fine, but an alternative you can use is to import a "minimal" example via code like this:
```
Procedure procedure = TestUtil.readExampleResource("json/ibm/minimal/Procedure-1.json")
procedure = procedure.toBuilder().
        subject(Reference.builder()
            .reference(com.ibm.fhir.model.type.String.of("urn:1"))
            .build())
        .build()
```
Its most useful if you have resource types with a lot of required fields.

Reply:
It might even be worth wrapping that pattern is a new helper method like `TestUtil.minimal(Procedure.class)` (which would return a new Builder for the passed resource type class, which has just the required fields populated).  Thoughts?

Reply:
I did create a new TestUtil.getMinimalResource() method, per your suggestion. However, fhir-server does not currently pull in the fhir-model test-jar, and when I tried updating the pom to do so, I was getting some 'class not found' exceptions, so decided to just go with what I have. We can revisit if you think it's important.

Reply:
not important.  will be interested to see if we can make use of this new method over time...it seems generally useful to me; like a cleaner version of the pattern I described above